iot-hub/Samples/device/ConventionBasedOperations/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -496,5 +515,24 @@ private static double GenerateTemperatureWithinRange(int max = 50, int min = 0)
         {
             return Math.Round(s_random.NextDouble() * (max - min) + min, 1);
         }
+
+        private async Task UpdateForEmptyWritableProperty(string componentName, CancellationToken cancellationToken)
+        {
+            const string propertyName = "targetTemperature";
+            
+            var reportedProperties = new ClientPropertyCollection();
+
+            // If the writable properties are empty, report the default value with ACK(ac=203, av=0) as part of the PnP convention.
+            // "DefaultPropertyValue" is set from the device when the desired property is not set via the hub.
+            var propertyValue = _deviceClient.PayloadConvention.PayloadSerializer.CreateWritablePropertyResponse(
+                DefaultPropertyValue, 203, 0);

```

Comment:
Instead of passing in the status code directly, I'd recommend extending `CommonClientResponseCodes` and adding in the application specific codes with the doc comments. This will guide users to adopt the recommended practices.